src/storage/mongodb/ChargingStationStorage.ts

```diff
@@ -732,12 +741,29 @@ export default class ChargingStationStorage {
     aggregation.push({
       $limit: dbParams.limit
     });
+    if (params.withChargingStation || params.withSiteArea) {
+      DatabaseUtils.pushChargingStationLookupInAggregation({
+        tenantID, aggregation, localField: 'chargingStationID', foreignField: '_id',
+        asField: 'chargingStation',
+        oneToOneCardinality: true,
+        oneToOneCardinalityNotNull: false
+      });
+    }
     // Project
     DatabaseUtils.projectFields(aggregation, projectFields);
     // Read DB
     const chargingProfilesMDB = await global.database.getCollection<ChargingProfile>(tenantID, 'chargingprofiles')
       .aggregate(aggregation, { collation: { locale: Constants.DEFAULT_LOCALE, strength: 2 }, allowDiskUse: true })
       .toArray();
+
+    if (params.withSiteArea) {
+      for (const chargingProfile of chargingProfilesMDB) {
+        if (chargingProfile.chargingStation.siteAreaID) {
+          chargingProfile.siteArea = await SiteAreaStorage.getSiteArea(tenantID, chargingProfile.chargingStation.siteAreaID);
```

Comment:
@mahdibenromdhane  
This is difficult in this case, because the SiteAreaID is not part of a chargingProfile. Thats why I have to get the ChargingStation with the siteAreaID first.

Reply:
Site Area ID is part of the charging station and can be requested with a lookup, I'll change it.